File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercebomentry/CommercebomentryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercebomentry;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercebomentryService extends BaseService {

	public CommercebomentryService(Session session) {
		super(session);
	}

	public JSONObject addCommerceBomEntry(long userId, int number, String cpInstanceUuid, long cProductId, long commerceBOMDefinitionId, double positionX, double positionY, double radius) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMDefinitionId", commerceBOMDefinitionId);
			_params.put("cpInstanceUuid", checkNull(cpInstanceUuid));
			_params.put("cProductId", cProductId);
			_params.put("number", number);
			_params.put("positionX", positionX);
			_params.put("positionY", positionY);
			_params.put("radius", radius);
			_params.put("userId", userId);

			_command.put("/commerce.commercebomentry/add-commerce-bom-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceBomEntry(long commerceBOMEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMEntryId", commerceBOMEntryId);

			_command.put("/commerce.commercebomentry/delete-commerce-bom-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONArray getCommerceBomEntries(long commerceBOMDefinitionId, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMDefinitionId", commerceBOMDefinitionId);
			_params.put("end", end);
			_params.put("start", start);

			_command.put("/commerce.commercebomentry/get-commerce-bom-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceBomEntriesCount(long commerceBOMDefinitionId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMDefinitionId", commerceBOMDefinitionId);

			_command.put("/commerce.commercebomentry/get-commerce-bom-entries-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject getCommerceBomEntry(long commerceBOMEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMEntryId", commerceBOMEntryId);

			_command.put("/commerce.commercebomentry/get-commerce-bom-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceBomEntry(long commerceBOMEntryId, int number, String cpInstanceUuid, long cProductId, double positionX, double positionY, double radius) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceBOMEntryId", commerceBOMEntryId);
			_params.put("cpInstanceUuid", checkNull(cpInstanceUuid));
			_params.put("cProductId", cProductId);
			_params.put("number", number);
			_params.put("positionX", positionX);
			_params.put("positionY", positionY);
			_params.put("radius", radius);

			_command.put("/commerce.commercebomentry/update-commerce-bom-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Commercebomentry/CommercebomentryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercebomentry
{
	public class CommercebomentryService : ServiceBase
	{
		public CommercebomentryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddCommerceBomEntryAsync(long userId, int number, string cpInstanceUuid, long cProductId, long commerceBOMDefinitionId, double positionX, double positionY, double radius)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("number", number);
			_parameters.Add("cpInstanceUuid", cpInstanceUuid);
			_parameters.Add("cProductId", cProductId);
			_parameters.Add("commerceBOMDefinitionId", commerceBOMDefinitionId);
			_parameters.Add("positionX", positionX);
			_parameters.Add("positionY", positionY);
			_parameters.Add("radius", radius);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/add-commerce-bom-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceBomEntryAsync(long commerceBOMEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceBOMEntryId", commerceBOMEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/delete-commerce-bom-entry", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<IEnumerable<dynamic>> GetCommerceBomEntriesAsync(long commerceBOMDefinitionId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceBOMDefinitionId", commerceBOMDefinitionId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/get-commerce-bom-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceBomEntriesCountAsync(long commerceBOMDefinitionId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceBOMDefinitionId", commerceBOMDefinitionId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/get-commerce-bom-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceBomEntryAsync(long commerceBOMEntryId, int number, string cpInstanceUuid, long cProductId, double positionX, double positionY, double radius)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceBOMEntryId", commerceBOMEntryId);
			_parameters.Add("number", number);
			_parameters.Add("cpInstanceUuid", cpInstanceUuid);
			_parameters.Add("cProductId", cProductId);
			_parameters.Add("positionX", positionX);
			_parameters.Add("positionY", positionY);
			_parameters.Add("radius", radius);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/update-commerce-bom-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceBomEntryAsync(long commerceBOMEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceBOMEntryId", commerceBOMEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercebomentry/get-commerce-bom-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}